pkg/virt-controller/services/vm.go

```diff
@@ -63,11 +64,18 @@ type vmService struct {
 }
 
 func (v *vmService) StartVMPod(vm *corev1.VM) error {
-
 	precond.MustNotBeNil(vm)
 	precond.MustNotBeEmpty(vm.GetObjectMeta().GetName())
+	precond.MustNotBeEmpty(vm.GetObjectMeta().GetNamespace())
 	precond.MustNotBeEmpty(string(vm.GetObjectMeta().GetUID()))
 
+	err := registrydisk.Initialize(vm, v.KubeCli)
```

Comment:
Yes - Initialization is okay, but it should not be tied to the storage method.


I.e. I do see that we want to attach secrets in this step, but this should be about attaching the secret - and the registry disk is just one thing consuming it later on.